src/altinn-app-frontend/src/types/index.ts

```diff
@@ -160,14 +164,41 @@ export interface IUiConfig {
   repeatingGroups?: IRepeatingGroups;
   fileUploadersWithTag?: IFileUploadersWithTag;
   navigationConfig?: INavigationConfig;
-  layoutOrder: string[];
+  tracks: ITracks;
   pageTriggers?: Triggers[];
   hideCloseButton?: boolean;
   showLanguageSelector?: boolean;
   showProgress?: boolean;
   keepScrollPos?: IKeepComponentScrollPos;
 }
 
+/**
+ * This state includes everything needed to calculate which layouts should be shown, and their order.
+ * @see https://docs.altinn.studio/app/development/ux/pages/tracks/
+ */
+export interface ITracks {
+  /**
+   * The main 'order' is the list of layouts available, or which layouts the server tells us to display. If a layout
+   * is not in this list, it should be considered hidden. It will be null until layouts have been fetched.
+   *
+   * Do NOT use this directly, as it will not respect layouts hidden using expressions!
+   * @see getLayoutOrderFromTracks
+   * @see selectLayoutOrder
+   */
+  order: string[] | null;
+
+  /**
+   * This state contains the results from calculating `hiddenExpr` (expressions to decide if a certain layout should
+   * be hidden or not). If a layout is in this list, is should also not be displayed.
+   */
+  hidden: string[];
+
+  /**
+   * List of expressions containing logic used to show/hide certain layouts.
+   */
+  hiddenExpr: IHiddenLayoutsExpressions;
```

Comment:
Could `hidden` and `hiddenExpr` be nullable so that one does not have to include a empty list and an empty expression object when hidden functionality is not wanted?

Reply:
I guess, yes, but i'd rather include these empty lists in tests (which would fail loudly without them) than somehow forget to add them to production code not covered by tests (where it would fail silently/weirdly, or just crash the app). I think nullability is evil as long as we have set up `"strictNullChecks": false` in our `tsconfig.json`.

Reply:
Btw, I had that on my TODO-list, but reported it now: #556